src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(stateClone, action.extraData);
```

Comment:
Consider renaming 'extraData' to a more descriptive name that reflects its purpose.

Reply:
How can I rename it if it's a key of the object? Do you mean creating a new const?

Reply:
@yslyshchenko ah I see, never mind then)